addons/html_editor/controllers/main.py

```diff
@@ -547,3 +562,47 @@ def bus_broadcast(self, model_name, field_name, res_id, bus_data):
         bus_data.update({'model_name': model_name, 'field_name': field_name, 'res_id': res_id})
         request.env['bus.bus']._sendone(channel, 'editor_collaboration', bus_data)
 
+    @http.route('/html_editor/link_preview_external', type="json", auth="public", methods=['POST'])
```

Comment:
I'm surprised we need this route, I may be wrong but I think it's best for the client to fetch the metadata himself.

Reply:
@fdardenne I tried to fetch the external links directly in the frontend at the beginning, but usually it receives an error because of the CORS policy. That's why I used the back-end for it. But maybe there's a way to do it at the front-end properly? :pray: 

@xmo-odoo If it's necessary and should be in `mail`, shall I create a new file in the mail controller and add it? Sorry for the naive question, just wanna be careful when touching another model :pray:

Reply:
My bad @Jinjiu96, you were right. I just tested how other known social network does and they do not let the browser fetch the preview. 
Security wise it seems logic after all, we don't want Bob to put his "malicous" website in the editor and get all the browser fingerprint informations about each reader previewing the link.